source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -105,6 +132,77 @@ Extractor::extractValue(Http::StreamFilterCallbacks &callbacks,
   return "";
 }
 
+// Match a regex against the input value and replace the matched subgroup with the replacement_text_ value
+// writes the result to replaced_value_ and returns a absl::string_view to it
+absl::string_view
+Extractor::replaceIndividualValue(Http::StreamFilterCallbacks &callbacks,
+                                  absl::string_view value) const {
+  std::match_results<absl::string_view::const_iterator> regex_result;
+
+  // if there are no matches, return an empty string
+  if (!std::regex_search(value.begin(), value.end(), regex_result, extract_regex_)) {
+    ENVOY_STREAM_LOG(debug, "replaceValue: extractor regex did not match input", callbacks);
+    return "";
+  }
+
+  // if the subgroup specified is greater than the number of subgroups in the regex, return an empty string
+  if (group_ >= regex_result.size()) {
+    ASSERT("no such group in the regex");
```

Comment:
curious, what does `ASSERT`ing a string literal do here?

Reply:
I think we'll segfault if we hit this... I don't think we should ever hit this (maybe in the no match case 😬) but I will confirm and do something else if that is in fact the case

Reply:
Some notes:
 - I lifted this block from pre-existing extractor code, in `Extractor::extractValue`
https://github.com/solo-io/envoy-gloo/blob/f747a124c15c56a09d6436607f4efc992e37859d/source/extensions/filters/http/transformation/inja_transformer.cc#L121-L126
 - I have confirmed that the comment in the pre-existing code indicating that this block should never be entered. 
   - Specifically, the new test `AttemptReplaceFromPartialMatchNonNilSubgroup` validates this -- it confirms that `regex_result.size()` is equal to the number of capturing groups in the regex plus 1, even in the case where the regex does not match
https://github.com/solo-io/envoy-gloo/blob/f747a124c15c56a09d6436607f4efc992e37859d/test/extensions/filters/http/transformation/inja_transformer_replace_test.cc#L153-L174
 - I have added another test to validate this here: 0f64ffe89d845bd8f2242966033730165f3bdd50
 - I have updated the comment in the block you highlighted to match the comment from the pre-existing code, indicating that this block should never be reached as this condition is validated in the constructor: 1440a95b8a3ff6ad03c1d49c82b9d2f1e1732bb0